pipeline-runner/R/gem2s-2-load_user_files.R

```diff
@@ -44,6 +45,52 @@ load_user_files <- function(input, pipeline_config, prev_out, input_dir = INPUT_
   return(res)
 }
 
+#' Read h5 file
+#'
+#' Calls read10x_h5
+#'
+#' @param config
+#' @param input_dir
+#'
+#' @return
+#' @export
+#'
+#' @examples
+read_10x_h5_file <- function(config, input_dir) {
+  counts_list <- list()
+  annot_list <- list()
+
+  samples <- config$samples
+
+  for (sample in samples) {
+    sample_dir <- file.path(input_dir, sample)
+    sample_fpaths <- list.files(sample_dir)
+    sample_counts_path <- file.path(sample_dir, sample_fpaths[[1]])
+
+    message("\nSample --> ", sample)
+    message("Reading files from ",
+            sample_dir,
+            " --> ",
+            paste(sample_fpaths, collapse = " - "))
+
+    if (length(sample_fpaths) > 1)
+      stop("Only one h5 expected. More files detected.")
+
+    ungzipped_counts_path <- R.utils::gunzip(sample_counts_path)
+
+    gene_names <- rownames(Seurat::Read10X_h5(ungzipped_counts_path))
+    counts <- Seurat::Read10X_h5(ungzipped_counts_path, use.names = F)
```

Comment:
This fixes a bug where an H5 pipeline will fail if there are multiple modalities. To test, you can use [these H5 files](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE183716) which are from the workshop dataset

```suggestion
    counts_names <- Seurat::Read10X_h5(ungzipped_counts_path)
    counts <- Seurat::Read10X_h5(ungzipped_counts_path, use.names = FALSE)

    # use Gene Expression modality if multiple
    if (methods::is(counts, 'list')) {
      counts_names <- counts_names$`Gene Expression`
      counts <- counts$`Gene Expression`
    }

    gene_names <- row.names(counts_names)
```